### PR TITLE
Extended stack

### PR DESCRIPTION
**Pull request by mattcaron-token**
This is Matt, driving some stuff for Taylor.

This is Taylor's `extended_stack` branch, containing changes to be used with the "extended" STM BLE stack.

This work generally falls into 3 categories:
1. Saving space.
2. Adjusting config for local artifactory, etc.
3. Actual bug fixes.

And they are all co-mingled. Have fun!

### CODE REVIEW

File: src/types/advertisement.rs
Comment (ghost) [inline src/types/advertisement.rs:93]: I think using `core::mem::sizeof::<u16>` and `core::mem::sizeof::<u128>` might be better than using magic numbers.

File: src/vendor/command/l2cap.rs
Comment (ghost) [inline src/vendor/command/l2cap.rs:372]: There's no explicit check if direct indexing `channel_index_list[..channel_number]` is valid. Probably this is implied somewhere else?

File: src/vendor/command/l2cap.rs
Comment (ghost) [inline src/vendor/command/l2cap.rs]: It looks like we should be asserting for `MIN_LENGHT`?

File: src/vendor/event/mod.rs
Comment (ghost) [inline src/vendor/event/mod.rs]: Shouldn't this be `13`?

**Comment by TaylorCarpenter** [comment]
I dropped the commit relating to size reduction, since it didn't do much and was premature.

File: src/vendor/command/l2cap.rs
Comment (TaylorCarpenter) [reply src/vendor/command/l2cap.rs]: Since this is a static assertion, the byte array we are copying into must be large enough to support the maximum payload size.

File: src/vendor/event/mod.rs
Comment (TaylorCarpenter) [reply src/vendor/event/mod.rs]: Answer is "no, because this is currently working so the length matches". But why do you think it should be 13 (and maybe by the time you answer, I can remember why I have it at 11...)?

Comment (ghost) [reply src/vendor/event/mod.rs]: A few lines after 
```
num_channels: buffer[12] })
```
It works because 13 is more than 11, but if you would pass a 12 or 11 it would fail in the wrong spot.

File: src/vendor/command/hal.rs
Comment (mattcaron-token) [inline src/vendor/command/hal.rs]: Note: I need this for the FCC test, but I already have a PR ready to go once this is merged which restores what I need.

(Feel free to close once read).

File: src/vendor/command/l2cap.rs
Comment (TaylorCarpenter) [reply src/vendor/command/l2cap.rs]: Never mind... It is not a static assertion and we use min/max elsewhere...

File: src/types/advertisement.rs
Comment (TaylorCarpenter) [reply src/types/advertisement.rs:93]: I agree, but this is remaining consistent with the rest of the codebase. It would be a little strange to use that here but then have magic numbers everywhere else.

**Review by mattcaron-token — COMMENTED** [review]
Apart from what Alessandro noted, this looks fine to me.

File: src/vendor/command/hal.rs
Comment (mattcaron-token) [reply src/vendor/command/hal.rs]: This is no longer valid - it was dropped in a rebase. Closing.

**Comment by mattcaron-token** [comment]
I had a branch which depended on this which restored things Taylor deleted. But, they're no longer deleted, so all the branch does is fix up the version number... which is silly to have all by itself with this in flight.

Since I can't approve this PR (as I am the author), I've just added it to this.

**Review by ghost — APPROVED** [review]

**Comment by ghost** [comment]
Can't seem to be able to resolve conversations (permissions?). All looks good to me now.

**Comment by DavidZemon** [comment]
> Can't seem to be able to resolve conversations (permissions?). All looks good to me now.

Sorry; fixed now

**Review by TaylorCarpenter — APPROVED** [review]

**Pull request merged**
